src/terminal/terminal.rs

```diff
@@ -256,6 +261,9 @@ where
         // and the terminal (if growing), which may OOB.
         self.autoresize()?;
 
+        // increment frame count before getting frame
+        self.frame_count = self.frame_count.wrapping_add(1);
+
```

Comment:
Would it make sense to increment after rendering? so the first frame is `0`?

Reply:
The intended the first frame to be 1 since I wanted to include the count of the current render into the frame count.

```
    /// This method provides access to the frame count, which is a sequence number indicating
    /// how many frames have been rendered up to and including this one. 
```

When a user calls `f.count()` in the first frame, I thought it made more sense to return `1` then. 

Also, if we increment after rendering, the `f.count()` will be `0` but `CompletedFrame`, i.e. `let completed_frame = terminal.draw(|f| {...});` will have `completed_frame.count` be `1`.

With this implementation both `f.count()` and `completed_frame.count` will be the same.

Thoughts?

Reply:
The second part of this is a non-issue as it's simple to avoid. Create the `CompletedFrame` and store it in a local variable before incrementing the value and then return the completed frame. So only the first part is relevant here.

In my mind I kinda have this idea that it's the index of the frame, not the number of frames rendered. I don't think either is the incorrect perspective. Though perhaps calling it index instead of count might make sense.

Where it might make (a very small) difference is in animations - if you have an animation that starts from frame 0 at the beginning of a program, then you'd need to subtract 1 from the frame count. Not sure if there are other calculations that would potentially have off by one bugs, so I guess it's no big deal.

Reply:
Should we rename it to frame index?

Reply:
I defer to you on this one. Either is fine and so I can't decide. You've go the use cases in mind for it though.

Reply:
I like starting at `0` because you'll the first element in the array when you use `f.count() % vector.len()`. I guess what's the best name for the variable and the method on the `Frame` struct for this.

I asked chatgpt for alternative name suggestions.

```
When considering alternative names for the count field in the Frame struct, it's important to choose a name that accurately reflects its purpose and usage in the context of frame rendering or tracking. Here are some alternative names that might be suitable, depending on the exact role and functionality of this field:

    frameNumber: This name clearly indicates that the field represents the sequence number of the current frame.

    renderCount: This name suggests that the field tracks the number of times rendering has occurred.

    sequenceIndex: This implies that the field is an index in a sequence, which in this case is the sequence of rendered frames.

    frameSequence: Similar to sequenceIndex, this indicates a sequential ordering of frames.

    renderSequence: This name emphasizes the rendering process in the sequence tracking.

    frameIndex: This straightforwardly indicates an index value for frames.

    iterationCount: If the field is used to count iterations of a rendering loop, this name would be appropriate.

    updateCount: This could be used if the field is incremented every time the frame is updated.

    frameId: This suggests a unique identifier for each frame, though 'ID' often implies a broader uniqueness than a simple count.

    renderIteration: This name combines the rendering process with the notion of iterative counts.
```

I like `f.count()` more than `f.number()` or `f.index()`.

Reply:
I'd be inclined to keep frame in there possibly (`frame_number`)? Frame number is pretty common when dealing with video / audio frames.

Reply:
Alternatively just accept that count is the count of frames previous to this one.

Reply:
I left the name as frame count but changed it to start from zero. I've updated the tests and the docstring too.